packages/theme/src/cli/utilities/theme-environment/hot-reload/client.ts

```diff
@@ -51,70 +68,95 @@ function hotReloadScript() {
     }
   }
 
-  let serverPid: string | undefined
-  const evtSource = new EventSource('/__hot-reload/subscribe', {withCredentials: true})
+  const refreshHTMLLinkElements = (elements: HTMLLinkElement[]) => {
+    for (const element of elements) {
+      // The `href` property prepends the host to the pathname. Use attributes instead:
+      element.setAttribute('href', element.getAttribute('href')!.replace(/v=\d+$/, `v=${Date.now()}`))
+    }
+  }
 
-  evtSource.onopen = () => logInfo('Connected')
-  evtSource.onerror = (event) => {
-    if (event.eventPhase === EventSource.CLOSED) {
-      logError('Connection closed by the server, attempting to reconnect...')
-    } else {
-      logError('Error occurred, attempting to reconnect...')
+  const refreshSections = async (data: HotReloadEvent & {type: 'section' | 'extAppBlock'}, elements: Element[]) => {
+    const controller = new AbortController()
+
+    await Promise.all(
+      elements.map(async (element) => {
+        const prefix = data.type === 'section' ? 'section' : 'app'
+        const sectionId = element.id.replace(/^shopify-section-/, '')
+        const params = [
+          `section-id=${encodeURIComponent(sectionId)}`,
+          `${prefix}-template-name=${encodeURIComponent(data.key)}`,
+          `pathname=${encodeURIComponent(window.location.pathname)}`,
+          `search=${encodeURIComponent(window.location.search)}`,
+        ].join('&')
+
+        const response = await fetch(`/__hot-reload/render?${params}`, {signal: controller.signal})
+
+        if (!response.ok) {
+          throw new Error(`Hot reload request failed: ${response.statusText}`)
+        }
+
+        const updatedSection = await response.text()
+
+        // eslint-disable-next-line require-atomic-updates
+        element.outerHTML = updatedSection
+      }),
+    ).catch((error: Error) => {
+      controller.abort('Request error')
+      fullPageReload(data.key, error)
+    })
+  }
+
+  const refreshAppEmbedBlock = async (data: HotReloadEvent & {type: 'extAppBlock'}, block: Element) => {
+    const controller = new AbortController()
+
+    const appBlockId = block.id.replace(/^shopify-block-/, '')
+    const params = [
+      `app-block-id=${encodeURIComponent(appBlockId)}`,
```

Comment:
Is this going to be used for all `app blocks` or just `app embeds`?
Related to [this comment](https://github.com/Shopify/cli/pull/4375/files#r1736867327)

Reply:
In this context, we're only handling app embeds, so it's indeed fair to rename this! :)